docs/guide/blog/comment-blog.md

```diff
@@ -0,0 +1,548 @@
+
+```go
+message MsgCreateComment {
+  string creator = 1;
+  int32 blogID = 2;
+  string title = 3;
+  string body = 4;
+  int64 createdAt = 5;
+}
+
+message MsgCreateCommentResponse {
+  uint64 id = 1;
+}
+```
+
+ The `MsgCreateComment` has five fields: creator, title, body, blogID and createdAt. Since the purpose of the `MsgCreateComment` message is to create new comments in the store, the only thing the message needs to return is an ID of a created comments. The `CreateComment` rpc was already added to the `Msg` service:
+
+```go
+  rpc CreateComment(MsgCreateComment) returns (MsgCreateCommentResponse);
+```
+
+Also, make a small modification in `MsgCreatePost` to add `createAt`
+
+```go
+message MsgCreatePost {
+  string creator = 1;
+  string title = 2;
+  string body = 3;
+  int64 createdAt = 4;
+}
+```
+
+Next, look at the `x/blog/handler.go` file. Starport has added a `case` to the `switch` statement inside the `NewHandler` function. This switch statement is responsible for routing messages and calling specific keeper methods based on the type of the message. `case *types.MsgCreateComment` has been added along with previously added `case *types.MsgCreatePost`
+
+```go
+func NewHandler(k keeper.Keeper) sdk.Handler {
+	//...
+		switch msg := msg.(type) {
+		//...
+		case *types.MsgCreateComment:
+			res, err := msgServer.CreateComment(sdk.WrapSDKContext(ctx), msg)
+			return sdk.WrapServiceResult(ctx, res, err)
+			//...
+		}
+	}
+}
+```
+
+The `case *types.MsgCreateComment` statement handles messages of type `MsgCreateComment`, calls the `CreateComment` method, and returns back the response.
+
+
+## Process Messages
+
+In the newly scaffolded `x/blog/keeper/msg_server_create_comment.go` file, you can see a placeholder implementation of the `CreateComment` function. Right now it does nothing and returns an empty response. For your blog chain, you want the contents of the message (title and body) to be written to the state as a new comment.
+
+You need to do two things:
+
+- Create a variable of type `Comment` with title and body from the message
+- Check if the the comment posted for the respective blog id exists and comment is not older than 100 blocks.
+- Append this `Comment` to the store
+
+```go
+import (
+    //...
+    sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+func (k msgServer) CreateComment(goCtx context.Context, msg *types.MsgCreateComment) (*types.MsgCreateCommentResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	BlockHeight := CreatedAt() // Invoke method to get the block height of Post
+	PostID := CreatedId() // Invoke method to get the Post ID
+
+	// Create variable of type comment
+	var comment = types.Comment{
+		Creator: msg.Creator,
+		Id:      msg.Id,
+		Body:    msg.Body,
+		Title:   msg.Title,
+		BlogID:  msg.BlogID,
+		CreatedAt: ctx.BlockHeight(),
+	}
+
+	// Check if the Post Exists for which a comment is being created
+	if comment.BlogID > PostID {
```

Comment:
This error should be handled above, when `GetPost` returns an error, you throw an error that a blog post with a given post ID can't be found.